### DISCUSSION TITLE
Client Side Caching

### DISCUSSION THREAD

**Discussion by hjaber** [solidjs/solid-start#837] [Q&A] [answered]
Is there a native way on SolidStart to cache `createServerData$` on the client side?

Maybe I was deploying incorrectly, but I noticed that my subsequent requests to routes that have already been loaded, would still block rendering pending `routeData`. There was a noticeable delay when I was using airplane wifi and clicking on a `<A>`

The solidjs [Hacker News Demo](https://hackernews-csr.ryansolid.workers.dev/) appears to have user comments cached despite making a network request on every visit. I realize this demo is using solidjs (not SolidStart) but that behavior is what I'm looking for, almost instant loading on subsequent visits.

**Comment by atk** [comment, marked as answer]
https://github.com/solidjs-community/solid-primitives/tree/main/packages/resource#makecache might be a simple solution for you.

**Reply by hjaber** [reply]
> https://github.com/solidjs-community/solid-primitives/tree/main/packages/resource#makecache might be a simple solution for you.

Thanks for building and sharing @atk! So it seems that there is no native caching for SolidStart.

This looks like what I'm looking for, I don't need an extensive library like `solid-query` for a lot of data fetches, I just want to cache my data fetches to provide better UI/UX.

Is this primitive designed to work with SolidStart's `routeData` & `createRouteData`?

**Comment by hjaber** [comment]
For some reason, the 2 year old version of the hacker news demo is much faster when re-visiting comment pages, it feels like it is cached but I do see a network request still being sent. When I throttle my connection to slow 3G, the 2 year old version still loads instantly while the newer SolidStart crawls. 

I cannot figure out why

* Solidjs Hacker News [Demo](https://github.com/solidjs/solid-hackernews/)
* SolidStart Hacker News [Demo](https://solid-hn-islands.netlify.app/)

**Reply by boehs** [reply]
> it feels like it is cached but I do see a network request still being sent. 

It is cached. Check the file size. You can disable caching in devtools.

Even so, the comparison here is not fair. One is proxing an API (in 40ms), and one is doing SSR (which seems to be taking 100-200ms)

https://solid-start-hn.netlify.app/ vs https://hackernews-csr.ryansolid.workers.dev/ is much more fair

**Reply by hjaber** [reply]
> > it feels like it is cached but I do see a network request still being sent. 
> 
> It is cached. Check the file size. You can disable caching in devtools.
> 
> Even so, the comparison here is not fair. One is proxing an API (in 40ms), and one is doing SSR (which seems to be taking 100-200ms)
> 
> https://solid-start-hn.netlify.app/ vs https://hackernews-csr.ryansolid.workers.dev/ is much more fair

How is the solid start without islands version caching?
